packages/components/src/utils/get-focusable-children.ts

```diff
@@ -25,5 +25,14 @@ const focusDisablingSelector = `:where(${[
 ].join(',')})`;
 
 export function getFocusableChildren(element: Element): NodeListOf<HTMLElement> {
-  return element.querySelectorAll(`& > ${focusableSelector}:not(${focusDisablingSelector})`);
+  const focusableChildren = element.querySelectorAll(
+    `${focusableSelector}:not(${focusDisablingSelector})`,
+  );
+
+  const visibleFocusableChildren = Array.from(focusableChildren).filter(child => {
+    const style = window.getComputedStyle(child.parentElement);
+    return style.display !== 'none';
+  });
+
+  return visibleFocusableChildren as unknown as NodeListOf<HTMLElement>;
```

Comment:
This typing is incorrect, you're turning the node list into an array on line 32 so the return type is an array of HTML elements (`HTMLElement[]`)